interactions/api/gateway/client.py

```diff
@@ -441,55 +443,35 @@ def _dispatch_event(self, event: str, data: dict) -> None:
 
                 _cache: "Storage" = self._http.cache[model]
 
-                if isinstance(obj, Member):
-                    id = (Snowflake(data["guild_id"]), obj.id)
-                else:
-                    id = getattr(obj, "id", None)
-
+                id = ids = None
+                id = self.__get_object_id(data, obj, model)
                 if id is None:
-                    if model.__name__ == "GuildScheduledEventUser":
-                        id = model.guild_scheduled_event_id
-                    elif model.__name__ == "Presence":
-                        id = obj.user.id
-                    elif model.__name__ in [
-                        "Invite",
-                        "GuildBan",
-                        "ChannelPins",
-                        "MessageReaction",
-                        "MessageReactionRemove",
-                        "MessageDelete",
-                        # Extend this for everything that should not be cached
-                    ]:
-                        id = None
-                    elif model.__name__.startswith("Guild"):
-                        model_name = model.__name__[5:]
-                        if _data := getattr(obj, model_name, None):
-                            id = (
-                                getattr(_data, "id")
-                                if not isinstance(_data, dict)
-                                else Snowflake(_data["id"])
-                            )
-                        elif hasattr(obj, f"{model_name}_id"):
-                            id = getattr(obj, f"{model_name}_id", None)
+                    ids = self.__get_object_ids(obj, model)
 
                 def __modify_guild_cache():
```

Comment:
While you're optimizing, can you move `__modify_guild_cache` into its own function?

Reply:
Sure

Reply:
Done